src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/JsonArrayConverter.cs

```diff
@@ -10,7 +10,12 @@ internal sealed class JsonArrayConverter : JsonConverter<JsonArray>
     {
         public override void Write(Utf8JsonWriter writer, JsonArray value, JsonSerializerOptions options)
         {
-            Debug.Assert(value != null);
+            if (value is null)
```

Comment:
do we need to update nullable annotations?

Reply:
The `T value` parameter in `JsonConverter<T>.Write` is annotation agnostic (because it differs dynamically depending on the value of `HandleNull`) so it can't be made nullable.

Reply:
But shouldn't L9 be `internal sealed class JsonArrayConverter : JsonConverter<JsonArray?>`, since this can be used with nullable JsonArrays?